build.zig

```diff
@@ -23,52 +23,94 @@ pub fn build(b: *std.Build) void {
 
     const ssl = boringssl.artifact("ssl");
     const crypto = boringssl.artifact("crypto");
-    // const lshpack_c = lshpack_dep.path("lshpack.c");
-    // const lsqpack_c = lsqpack_dep.path("lsqpack.c");
 
-    const lib = b.addLibrary(.{
+    const lib = b.addStaticLibrary(.{
         .name = "lsquic",
-        .linkage = .static,
-
-        .root_module = b.createModule(
-            .{
-                .target = target,
-                .optimize = optimize,
-                .link_libc = true,
-            },
-        ),
+        .target = target,
+        .optimize = optimize,
     });
 
+    var c_flags = std.ArrayList([]const u8).init(b.allocator);
+
+    c_flags.appendSlice(&.{
+        "-DLSQUIC_DEBUG_NEXT_ADV_TICK=1",
+        "-DLSQUIC_CONN_STATS=1",
+        "-DLSQUIC_DEVEL=1",
+        "-DLSQUIC_WEBTRANSPORT_SERVER_SUPPORT=1",
+        "-fno-sanitize=undefined",
+    }) catch @panic("OOM");
+
+    if (optimize == .Debug) {
+        c_flags.appendSlice(&.{ "-O0", "-g3" }) catch @panic("OOM");
+    } else {
+        c_flags.appendSlice(&.{ "-O3", "-g0" }) catch @panic("OOM");
+    }
+
+    if (target.result.os.tag == .windows) {
+        lib.addIncludePath(upstream.path("wincompat"));
+        c_flags.appendSlice(&.{
+            "/W4",                       "/WX",
+            "-DWIN32_LEAN_AND_MEAN",     "-DNOMINMAX",
+            "-D_CRT_SECURE_NO_WARNINGS", "/wd4100",
+            "/wd4115",                   "/wd4116",
+            "/wd4146",                   "/wd4132",
+            "/wd4200",                   "/wd4204",
+            "/wd4244",                   "/wd4245",
+            "/wd4267",                   "/wd4214",
+            "/wd4295",                   "/wd4324",
+            "/wd4334",                   "/wd4456",
+            "/wd4459",                   "/wd4706",
+            "/wd4090",                   "/wd4305",
```

Comment:
i'm not developing on windows so I can't test this; if this is tested i'd say feel free to merge, but otherwise we should refrain from adding flags we don't know whether works or not (unless you've tested this flow, or this is from somewhere; in which case, please add a link)

Reply:
Yes, not tested yet, only extract from the cmakelist, I can comment out it. We can add a CI in this repo and try to test on windows later.

Reply:
We can leave it in; I saw where you lifted it from. I think we can just leave the flags and add a permalink to the lines inside the `CMakeLists.txt`:

https://github.com/litespeedtech/lsquic/blob/70486141724f85e97b08f510673e29f399bbae8f/CMakeLists.txt#L102-L137